osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -88,6 +87,11 @@ public abstract partial class MultiplayerClient : Component, IMultiplayerClient,
         /// </summary>
         public event Action? ResultsReady;
 
+        /// <summary>
+        /// Invoked just prior to disconnection requested by the server via <see cref="IStatefulUserHubClient.DisconnectRequested"/>.
+        /// </summary>
+        public event Action? Disconnecting;
```

Comment:
Can probably move this to the interface?

```diff
diff --git a/osu.Game/Online/IStatefulUserHubClient.cs b/osu.Game/Online/IStatefulUserHubClient.cs
index 86105dd629..ec35be8ff7 100644
--- a/osu.Game/Online/IStatefulUserHubClient.cs
+++ b/osu.Game/Online/IStatefulUserHubClient.cs
@@ -1,6 +1,7 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using System.Threading.Tasks;
 
 namespace osu.Game.Online
@@ -13,6 +14,11 @@ namespace osu.Game.Online
     /// </summary>
     public interface IStatefulUserHubClient
     {
+        /// <summary>
+        /// Invoked just prior to disconnection requested by the server via <see cref="IStatefulUserHubClient.DisconnectRequested"/>.
+        /// </summary>
+        public event Action? Disconnecting;
+
         Task DisconnectRequested();
     }
 }
diff --git a/osu.Game/Online/Multiplayer/MultiplayerClient.cs b/osu.Game/Online/Multiplayer/MultiplayerClient.cs
index 79f46c2095..4d473a4b60 100644
--- a/osu.Game/Online/Multiplayer/MultiplayerClient.cs
+++ b/osu.Game/Online/Multiplayer/MultiplayerClient.cs
@@ -32,6 +32,8 @@ public abstract partial class MultiplayerClient : Component, IMultiplayerClient,
 
         public Action<Room, string>? PresentMatch { protected get; set; }
 
+        public event Action? Disconnecting;
+
         /// <summary>
         /// Invoked when any change occurs to the multiplayer room.
         /// </summary>
@@ -87,11 +89,6 @@ public abstract partial class MultiplayerClient : Component, IMultiplayerClient,
         /// </summary>
         public event Action? ResultsReady;
 
-        /// <summary>
-        /// Invoked just prior to disconnection requested by the server via <see cref="IStatefulUserHubClient.DisconnectRequested"/>.
-        /// </summary>
-        public event Action? Disconnecting;
-
         /// <summary>
         /// Whether the <see cref="MultiplayerClient"/> is currently connected.
         /// This is NOT thread safe and usage should be scheduled.
diff --git a/osu.Game/Online/Spectator/SpectatorClient.cs b/osu.Game/Online/Spectator/SpectatorClient.cs
index ca4ec52f4a..0cc78734df 100644
--- a/osu.Game/Online/Spectator/SpectatorClient.cs
+++ b/osu.Game/Online/Spectator/SpectatorClient.cs
@@ -70,9 +70,6 @@ public abstract partial class SpectatorClient : Component, ISpectatorClient
         /// </summary>
         public virtual event Action<int, long>? OnUserScoreProcessed;
 
-        /// <summary>
-        /// Invoked just prior to disconnection requested by the server via <see cref="IStatefulUserHubClient.DisconnectRequested"/>.
-        /// </summary>
         public event Action? Disconnecting;
 
         /// <summary>

```

Reply:
I had it there but rider was graying it out, complaining that I ever only used it through the implementations, so I removed it... Will check again.

Reply:
It seems to hint that it's not used, but maybe fine to have there to enforce any future hubs follow standards..

Reply:
As per IRL discussion, the rider thing is less of an issue and the bigger issue is that apparently signalr doesn't like it and just refuses to have anything to do with any hub that has a client which declares an event. So this isn't happening.